# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/other/DocAccessibleChild.cpp
# Commit: 8d31c0604efe
# Full Hash: 8d31c0604efef2015995df32766d38dd4dd45ca2
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-08-04 04:34:13
# Regressor Bug: 1781536
# File Overlap Count: 1
# Description:
#   Bug 1781536 part 1: Support base Accessible in Relation. r=morgan
#   
#   Previously, Relation::Next returned a LocalAccessible.
#   Now, it returns an Accessible so that it can support RemoteAccessible as well.
#   Relation::LocalNext has been added as a convenience to return a LocalAccessible when the caller is certain that only LocalAccessible is relevant.
# ==============================================================================

diff -r af7f9589f48c -r 8d31c0604efe accessible/ipc/other/DocAccessibleChild.cpp
--- a/accessible/ipc/other/DocAccessibleChild.cpp	Thu Aug 04 02:47:34 2022 +0300
+++ b/accessible/ipc/other/DocAccessibleChild.cpp	Wed Aug 03 23:58:50 2022 +0000
@@ -146,7 +146,7 @@
 
   auto type = static_cast<RelationType>(aType);
   Relation rel = acc->RelationByType(type);
-  while (LocalAccessible* target = rel.Next()) {
+  while (LocalAccessible* target = rel.LocalNext()) {
     aTargets->AppendElement(reinterpret_cast<uint64_t>(target->UniqueID()));
   }
 
@@ -157,7 +157,7 @@
                         nsTArray<RelationTargets>* aTargets) {
   Relation rel = aAcc->RelationByType(aType);
   nsTArray<uint64_t> targets;
-  while (LocalAccessible* target = rel.Next()) {
+  while (LocalAccessible* target = rel.LocalNext()) {
     targets.AppendElement(reinterpret_cast<uint64_t>(target->UniqueID()));
   }
 